cli/src/fmt_deno_error.rs

```diff
@@ -0,0 +1,174 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+// https://github.com/denoland/deno/blob/main/cli/fmt_errors.rs
```

Comment:
I initially began writing all of this myself by hand, but after getting somewhat stumped by all the edgecases one has to handle to print a simple error message, I've looked at the Deno CLI's source code and decided it might be smarter to just take their error formatter.

As far as I can tell, this formatter is completely independent from the actual Deno CLI, so there shouldn't be any issues with just copying their output. If this PR is accepted I'd maybe also follow up with an upstream PR to expose this functionality, since it clearly seems at least somewhat useful to other people.